client/components/app.jsx

```diff
@@ -59,10 +83,14 @@ class App extends Component {
     const { view, cart } = this.state;
     return (
       <div className="container bg-light">
-        <Header cartItemCount={cart.length} />
+        <Header cartItemCount={cart.length} setView={this.setView} />
         {view.name === 'catalog'
           ? <ProductList setView={this.setView} />
-          : <ProductDetails addToCard={this.addToCard} setView={this.setView} params={view.params} />}
+          : view.name === 'cart'
+            ? <CartSummary backTo={this.backTo} cart={cart} setView={this.setView} />
+            : view.name === 'checkout'
+              ? <CheckoutForm backTo={this.backTo} cart={cart} setView={this.setView} placeOrder={this.placeOrder} />
+              : <ProductDetails backTo={this.backTo} addToCard={this.addToCard} setView={this.setView} params={view.params} />}
```

Comment:
Please do not nest ternaries. You can define a method on the `App` that returns the correct view. In that method, you can use a basic `switch` or `if/else`

Reply:
changed it~